public/components/overview/components/select_dashboard_flyout.tsx

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  EuiComboBox,
+  EuiComboBoxOptionOption,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFlyout,
+  EuiFlyoutBody,
+  EuiFlyoutFooter,
+  EuiFlyoutHeader,
+  EuiSmallButton,
+  EuiSmallButtonEmpty,
+  EuiText,
+} from '@elastic/eui';
+import React, { useEffect, useState } from 'react';
+import { uiSettingsService } from '../../../../common/utils';
+import { DashboardDictionary } from '../home';
+
+export interface Props {
+  closeFlyout: () => void;
+  wrapper: { dashboardSelected: boolean };
+  dashboards: DashboardDictionary;
+  registerDashboard: () => void;
+}
+
+export function SelectDashboardFlyout({
+  closeFlyout,
+  wrapper,
+  dashboards,
+  registerDashboard,
+}: Props) {
+  const [selectedOptionsState, setSelectedOptionsState] = useState<
+    Array<EuiComboBoxOptionOption<string>>
+  >([]);
+  const [selectedId, setSelectedId] = useState<string>('');
+  const [buttonIsActive, setButtonIsActive] = useState(false);
+
+  const onComboBoxChange = (options: Array<EuiComboBoxOptionOption<string>>) => {
+    if (options.length > 0) {
+      setButtonIsActive(options[0].value !== selectedId);
+    }
+    setSelectedOptionsState(options);
+  };
+
+  const onClickAdd = () => {
+    if (selectedOptionsState.length > 0) {
+      wrapper.dashboardSelected = true;
+      uiSettingsService
+        .set('observability:defaultDashboard', selectedOptionsState[0].value)
+        .then(registerDashboard);
+    }
+    closeFlyout();
+  };
+
+  useEffect(() => {
+    setSelectedId(uiSettingsService.get('observability:defaultDashboard'));
```

Comment:
Let's declare the settings key in common/constants

Reply:
Replaced with 
import { uiSettingsKey } from '../../../../common/constants/overview'
setSelectedId(uiSettingsService.get(uiSettingsKey));